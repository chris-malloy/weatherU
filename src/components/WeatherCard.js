import React from 'react';
import PropTypes from 'prop-types';
import { CardPanel, Row, Col } from 'react-materialize';

const WeatherCard = (props) => {
  var iconUrl = `https://openweathermap.org/img/w/${props.icon}.png`;
  var jacketMsg;
  if(props.temp > 80) {
    jacketMsg = "Stay home, jacket!";
  } else if (props.temp > 60) {
    jacketMsg = "Okay, maybe a light jacket";
  } else if (props.temp > 40) {
    jacketMsg = "Oh, you gonna need a jacket today";
  } else if (props.temp > 20) {
    jacketMsg = "What, it's winter already???";
  } else if (props.temp > 0) {
    jacketMsg = "STAY INSIDE";
  };

  var code = props.code.toString() // to use string proto includes()

  const messageMap = {
    "2": "Oh no, thunder! Bring 5 umbrellas",
    "3": "Just a drizzle today!",
    "5": "Umbrella time, it's one of those days",
    "6": "Snow Day!!!",
    "7": "Theres' some weird stuff going on today",
    "8": "Happy Day! Leave the umbrella home",
    "90": "Just stay home it's crazy out there"
  }

  const setUmbrellaMessage = () => {
    for (var i in messageMap) {
      if (code.includes(i)) {
        return messageMap[i];
      };
    };
  };
  

  var umbrellaMsg = setUmbrellaMessage();

  return (
      <Col s={12} m={6} l={4}>
        <CardPanel className="grey lighten-3">
          <Row className="center">
            <Col s={6} className="current-temp">
              {props.temp}°
            </Col>
            <Col s={6}>
            <img src={iconUrl} alt="weather icon"/>
            </Col>
            <Col s={12}>
              <p>{jacketMsg}</p>
              <p>{umbrellaMsg}</p>
            </Col>
            <Col s={6} className="date">
            {props.date}
            </Col>
            <Col s={6} className="high-low">
              <Col s={12}>
              {props.high}°
              </Col>
              <Col s={12}>
                {props.low}°
              </Col>
            </Col>
          </Row>
        </CardPanel>
    </Col>
  );
};

WeatherCard.propTypes = {
  temp: PropTypes.number.isRequired,
  icon: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  high: PropTypes.number.isRequired,
  low: PropTypes.number.isRequired,
  code: PropTypes.number.isRequired 
}

export default WeatherCard;