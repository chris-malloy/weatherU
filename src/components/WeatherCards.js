import React from 'react';
import WeatherCard from './WeatherCard';

const WeatherCards = (props) => {
  return props.weatherList.map((data, i) => {
    var item = data.list[0];
    console.log(item);
    return <WeatherCard key={i} temp={item.main.temp} icon={item.weather[0].icon} date={item.dt_txt} high={item.main.temp_max} low={item.main.temp_min} />
  });
};

export default WeatherCards;