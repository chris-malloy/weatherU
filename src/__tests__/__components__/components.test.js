import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import WeatherCards from '../../components/WeatherCards';
import WeatherCard from '../../components/WeatherCard'

configure({ adapter: new Adapter() });

describe('Components', () => {
  describe('WeatherCards', () => {
    const weatherCards = shallow(<WeatherCards />);

    it('renders correctly', () => {
      expect(weatherCards).toMatchSnapshot();
    });

    it('to have a `WeatherCard` component', () => {
      expect(weatherCards.find('WeatherCard').exists()).toBe(true);
    })
  });

  describe('WeatherCard', () => {
    const weatherCard = shallow(<WeatherCard />);

    it('renders correctly', () => {
      expect(weatherCard).toMatchSnapshot();
    });
  });
});