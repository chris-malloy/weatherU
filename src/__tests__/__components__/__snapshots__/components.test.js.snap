// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components WeatherCard renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherCard
    date="date"
    high={0}
    icon="icon"
    low={0}
    temp={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        0,
        "icon",
        "date",
        0,
        0,
      ],
    },
    "ref": null,
    "rendered": Array [
      0,
      "icon",
      "date",
      0,
      0,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          0,
          "icon",
          "date",
          0,
          0,
        ],
      },
      "ref": null,
      "rendered": Array [
        0,
        "icon",
        "date",
        0,
        0,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
